persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -456,7 +457,12 @@ public List<PolarisBaseEntity> lookupEntities(
     if (entityIds == null || entityIds.isEmpty()) return new ArrayList<>();
     PreparedQuery query = QueryGenerator.generateSelectQueryWithEntityIds(realmId, entityIds);
     try {
-      return datasourceOperations.executeSelect(query, new ModelEntity());
+      Map<PolarisEntityId, PolarisBaseEntity> idMap =
+          datasourceOperations.executeSelect(query, new ModelEntity()).stream()
+              .collect(
+                  Collectors.toMap(
+                      e -> new PolarisEntityId(e.getCatalogId(), e.getId()), Function.identity()));
+      return entityIds.stream().map(idMap::get).collect(Collectors.toList());
```

Comment:
Since we're obviously enforcing the order of the returned values to match the order of requested IDs, could you add a test for that?

Reply:
nevermind, I think existing tests are good enough :)